src/rules/prefer-native-locators.ts

```diff
@@ -0,0 +1,112 @@
+import { AST } from 'eslint'
+import { getStringValue, isPageMethod } from '../utils/ast'
+import { createRule } from '../utils/createRule'
+
+type Pattern = {
+  messageId: string
+  pattern: RegExp
+  replacement: string
+}
+
+export default createRule({
+  create(context) {
+    const { testIdAttribute } = {
+      testIdAttribute: 'data-testid',
+      ...((context.options?.[0] as Record<string, unknown>) ?? {}),
+    }
+
+    const patterns: Array<Pattern> = [
+      {
+        messageId: 'unexpectedLabelQuery',
+        pattern: /^\[aria-label=['"]?(.+?)['"]?\]$/,
+        replacement: 'getByLabel',
+      },
+      {
+        messageId: 'unexpectedRoleQuery',
+        pattern: /^\[role=['"]?(.+?)['"]?\]$/,
+        replacement: 'getByRole',
+      },
+      {
+        messageId: 'unexpectedPlaceholderQuery',
+        pattern: /^\[placeholder=['"]?(.+?)['"]?\]$/,
+        replacement: 'getByPlaceholder',
+      },
+      {
+        messageId: 'unexpectedAltTextQuery',
+        pattern: /^\[alt=['"]?(.+?)['"]?\]$/,
+        replacement: 'getByAltText',
+      },
+      {
+        messageId: 'unexpectedTitleQuery',
+        pattern: /^\[title=['"]?(.+?)['"]?\]$/,
+        replacement: 'getByTitle',
+      },
+      {
+        messageId: 'unexpectedTestIdQuery',
+        pattern: new RegExp(`^\\[${testIdAttribute}=['"]?(.+?)['"]?\\]`),
+        replacement: 'getByTestId',
+      },
+    ]
+
+    return {
+      CallExpression(node) {
+        if (node.callee.type !== 'MemberExpression') return
+        const query = getStringValue(node.arguments[0])
+        const isLocator = isPageMethod(node, 'locator')
+        if (!isLocator) return
+
+        // If it's something like `page.locator`, just replace the `.locator` part
+        const start =
+          node.callee.type === 'MemberExpression'
+            ? node.callee.property.range![0]
+            : node.range![0]
+        const end = node.range![1]
+        const rangeToReplace: AST.Range = [start, end]
```

Comment:
Move this into the `if (match)` only required if there is a match

Reply:
Good call, worth nothing that we should theoretically be able to get rid of the `MemberExpression` check, but after moving it into the fixer method, TypeScript loses the type inference so we need to keep it for now.